components/suggestion/switch_channel_provider.jsx

```diff
@@ -313,11 +316,18 @@ export function quickSwitchSorter(wrappedA, wrappedB) {
     if (b.type === Constants.DM_CHANNEL && bDisplayName.startsWith('@')) {
         bDisplayName = bDisplayName.substring(1);
     }
+    if (a.type === Constants.DM_CHANNEL && (b.type === Constants.DM_CHANNEL || b.type === Constants.GM_CHANNEL) && ((wrappedA.last_viewed_at && wrappedB.last_viewed_at) || (!wrappedA.last_viewed_at && !wrappedB.last_viewed_at))) {
```

Comment:
Does the new logic need to be this complicated? It already seemed a bit difficult to understand how this list is sorted, and I'm worried this makes that worse. What if instead of checking the channel types and whether they've been viewed, we just changed the below logic to 
```javascript
const aStartsWith = aDisplayName.startsWith(prefix) || wrappedA.name.toLowerCase().startsWith(prefix);
const bStartsWith = bDisplayName.startsWith(prefix) || wrappedB.name.toLowerCase().startsWith(prefix);
```
so that way, we're considering both the name and display name when prioritizing results matching the prefix

Reply:
@hmhealey I agree, yeah this should work.